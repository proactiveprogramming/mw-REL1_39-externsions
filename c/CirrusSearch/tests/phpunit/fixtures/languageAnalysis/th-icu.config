{
	"LangCode":"th",
	"withPlugins":["analysis-icu"]
}
