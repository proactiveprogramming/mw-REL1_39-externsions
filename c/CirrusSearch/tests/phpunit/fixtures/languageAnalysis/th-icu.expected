{
    "analyzer": {
        "text": {
            "type": "custom",
            "tokenizer": "icu_tokenizer",
            "char_filter": [
                "nnbsp_norm",
                "thai_repl_pat",
                "dotted_I_fix",
                "thai_charfilter"
            ],
            "filter": [
                "homoglyph_norm",
                "icu_normalizer",
                "decimal_digit",
                "thai_stop",
                "asciifolding"
            ]
        },
        "plain": {
            "type": "custom",
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer"
            ]
        },
        "plain_search": {
            "type": "custom",
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer"
            ]
        },
        "short_text": {
            "type": "custom",
            "tokenizer": "whitespace",
            "filter": [
                "icu_normalizer",
                "aggressive_splitting",
                "asciifolding_preserve",
                "dedup_asciifolding"
            ]
        },
        "short_text_search": {
            "type": "custom",
            "tokenizer": "whitespace",
            "filter": [
                "icu_normalizer",
                "aggressive_splitting"
            ]
        },
        "source_text_plain": {
            "type": "custom",
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer"
            ],
            "char_filter": [
                "word_break_helper_source_text"
            ]
        },
        "source_text_plain_search": {
            "type": "custom",
            "char_filter": [
                "word_break_helper_source_text"
            ],
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer"
            ]
        },
        "suggest": {
            "type": "custom",
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer",
                "suggest_shingle"
            ]
        },
        "suggest_reverse": {
            "type": "custom",
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer",
                "suggest_shingle",
                "reverse"
            ]
        },
        "token_reverse": {
            "type": "custom",
            "tokenizer": "no_splitting",
            "filter": [
                "reverse"
            ]
        },
        "near_match": {
            "type": "custom",
            "char_filter": [
                "near_space_flattener"
            ],
            "tokenizer": "no_splitting",
            "filter": [
                "icu_normalizer"
            ]
        },
        "near_match_asciifolding": {
            "type": "custom",
            "char_filter": [
                "near_space_flattener"
            ],
            "tokenizer": "no_splitting",
            "filter": [
                "truncate_keyword",
                "icu_normalizer",
                "asciifolding"
            ]
        },
        "prefix": {
            "type": "custom",
            "char_filter": [
                "near_space_flattener"
            ],
            "tokenizer": "prefix",
            "filter": [
                "icu_normalizer"
            ]
        },
        "prefix_asciifolding": {
            "type": "custom",
            "char_filter": [
                "near_space_flattener"
            ],
            "tokenizer": "prefix",
            "filter": [
                "icu_normalizer",
                "asciifolding"
            ]
        },
        "word_prefix": {
            "type": "custom",
            "tokenizer": "standard",
            "filter": [
                "icu_normalizer",
                "prefix_ngram_filter"
            ]
        },
        "keyword": {
            "type": "custom",
            "tokenizer": "no_splitting",
            "filter": [
                "truncate_keyword"
            ]
        },
        "lowercase_keyword": {
            "type": "custom",
            "tokenizer": "no_splitting",
            "filter": [
                "truncate_keyword",
                "icu_normalizer"
            ]
        },
        "trigram": {
            "type": "custom",
            "tokenizer": "trigram",
            "filter": [
                "icu_normalizer"
            ]
        },
        "text_search": {
            "type": "custom",
            "tokenizer": "icu_tokenizer",
            "char_filter": [
                "nnbsp_norm",
                "thai_repl_pat",
                "dotted_I_fix",
                "thai_charfilter"
            ],
            "filter": [
                "homoglyph_norm",
                "icu_normalizer",
                "decimal_digit",
                "thai_stop",
                "asciifolding"
            ]
        }
    },
    "filter": {
        "suggest_shingle": {
            "type": "shingle",
            "min_shingle_size": 2,
            "max_shingle_size": 3,
            "output_unigrams": true
        },
        "lowercase": {
            "type": "lowercase"
        },
        "aggressive_splitting": {
            "type": "word_delimiter",
            "stem_english_possessive": false,
            "preserve_original": false
        },
        "prefix_ngram_filter": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "asciifolding": {
            "type": "asciifolding",
            "preserve_original": false
        },
        "asciifolding_preserve": {
            "type": "asciifolding",
            "preserve_original": true
        },
        "truncate_keyword": {
            "type": "truncate",
            "length": 5000
        },
        "remove_empty": {
            "type": "length",
            "min": 1
        },
        "icu_normalizer": {
            "type": "icu_normalizer",
            "name": "nfkc_cf"
        },
        "thai_stop": {
            "type": "stop",
            "stopwords": "_thai_"
        },
        "dedup_asciifolding": {
            "type": "unique",
            "only_on_same_position": true
        }
    },
    "tokenizer": {
        "prefix": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "no_splitting": {
            "type": "keyword"
        },
        "trigram": {
            "type": "nGram",
            "min_gram": 3,
            "max_gram": 3
        }
    },
    "char_filter": {
        "near_space_flattener": {
            "type": "mapping",
            "mappings": [
                "'=>\\u0020",
                "\\u2019=>\\u0020",
                "\\u02BC=>\\u0020",
                "_=>\\u0020",
                "-=>\\u0020"
            ]
        },
        "nnbsp_norm": {
            "type": "mapping",
            "mappings": [
                "\\u202F=>\\u0020"
            ]
        },
        "word_break_helper": {
            "type": "mapping",
            "mappings": [
                "_=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020"
            ]
        },
        "word_break_helper_source_text": {
            "type": "mapping",
            "mappings": [
                "_=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020",
                ":=>\\u0020"
            ]
        },
        "dotted_I_fix": {
            "type": "mapping",
            "mappings": [
                "\u0130=>I"
            ]
        },
        "thai_repl_pat": {
            "type": "pattern_replace",
            "pattern": "(?<=\\p{Nd})([\u0e01-\u0e4f])|(?<=[\u0e01-\u0e4f])(\\p{Nd})|(?<=[\u0e01-\u0e4f])\\.([\u0e01-\u0e4f])",
            "replacement": " $1$2$3"
        },
        "thai_charfilter": {
            "type": "mapping",
            "mappings": [
                "_=>\\u0020",
                ";=>\\u0020",
                ":=>\\u0020",
                "\u00b7=>\\u0020",
                "\u2027=>\\u0020",
                "\u0e03=>\u0e02",
                "\u0e05=>\u0e04",
                "\\u0e4d\\u0e32=>\\u0e33",
                "\\u0e4d\\u0e48\\u0e32=>\\u0e48\\u0e33",
                "\\u0e4d\\u0e49\\u0e32=>\\u0e49\\u0e33",
                "\\u0e33\\u0e48=>\\u0e48\\u0e33",
                "\\u0e33\\u0e49=>\\u0e49\\u0e33",
                "\\u0E34\\u0E4D=>\\u0E36",
                "\\u0E4D\\u0E34=>\\u0E36"
            ]
        }
    }
}